component-runtime-beam/src/main/java/org/talend/sdk/component/runtime/beam/spi/record/AvroSchemaBuilder.java

```diff
@@ -339,6 +339,7 @@ public Schema build() {
                     .createRecord(SchemaIdGenerator.generateRecordName(avroFields), null, "talend.component.schema",
                             false);
             record.setFields(avroFields);
+            this.props.forEach(record::addProp);
             record.addProp(ENTRIES_ORDER_PROP, fields.stream().map(e -> e.getName()).collect(Collectors.joining(",")));
```

Comment:
I suggest this:
```java
record.addProp(ENTRIES_ORDER_PROP, fields.stream().map(e -> e.getName()).collect(Collectors.joining(",")));
props.entrySet()
         .stream()
         .filter(k -> !k.getKey().equals(ENTRIES_ORDER_PROP))
         .forEach(e -> record.addProp(e.getKey(), e.getValue()));
```
if `ENTRIES_ORDER_PROP` was inherited to avoid recdeclaring properties. (will throw an exception).

Reply:
done